### PR TITLE
fix: Gas Estimation failure displays 2 duplicated Warnings

### LINKED ISSUE #17866: [Bug]: Gas Estimation failure displays 2 duplicated Warnings
### Describe the bug

**Problem**: when there is a Gas Estimation failure, there are 2 duplicated warnings displayed on the Confirmation screen, saying `We were not able to estimate Gas...`;

Issue was introduced in 10.25.0

image

### Steps to reproduce

For a quick reproduction, we will use the incorrect behaviour (another bug is open for that), where when tx is going to fail, we display a huge amount of Gas and the warning unable to estimate gas.

https://user-images.githubusercontent.com/54408225/220713383-7cc42744-55db-4546-8562-ec516c3d9c45.mp4

1. Go to etherscan https://etherscan.io/address/0xe592427a0aece92de3edee1f18e0157c05861564/advanced#writeContract
2. Connect your MM acount
3. Click `selfPermit`
4. Fill it with values (0x, 0x for r and s)
5. Click Write
6. See MM prompt and 2 duplicated warnings

### Error messages or log output

_No response_

### Version

10.25.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by blackdevelopa**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23886?quickstart=1)

When there is a gas estimation failure, show only one gas warning on the confirmation screen. If the user proceeds to continue, show the gas component.

## **Related issues**

Fixes: #17866 

## **Manual testing steps**

1. Go to etherscan https://etherscan.io/address/0xe592427a0aece92de3edee1f18e0157c05861564/advanced#writeContract
2. Connect your MM acount
3. Click selfPermit
4. Fill it with values (0x, 0x for r and s)
5. Click Write
6. Should not see duplicated warnings

### **Before**

https://github.com/MetaMask/metamask-extension/assets/29962968/9fbee4e0-866b-4031-906e-ba2519163b3d

### **After**

https://github.com/MetaMask/metamask-extension/assets/29962968/c9e54305-e03f-42b0-923f-f8e4c641ab69

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by digiwand — COMMENTED** [review]
_edit: void comment_

**Review by digiwand — DISMISSED** [review]

**Review by cryptotavares — CHANGES_REQUESTED** [review]
I don't think that this is the right approach. We should take on this opportunity to remove the duplicated logic.. not render the same SimulationErrorMessage component in different places (at least not in `TransactionDetails` and `TransactionAlerts` within the ConfirmTransactionBase) depending on a prop.

If you remove the [this method to render the simulationFailureWarning](https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js#L515-L527), and rely solely on the `TransactionAlerts` component to render the alert, then you should be able to get a more slick experience (like on the recording below.. for example, the alert is always at the top, it does not move from the content to the top)

https://github.com/MetaMask/metamask-extension/assets/15957235/14002d29-aa0c-49fe-afcf-daf0f32dffa4

File: ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js [hunk 1]
```diff
@@ -576,6 +561,7 @@ export default class ConfirmTransactionBase extends Component {
           isBuyableChain={isBuyableChain}
           tokenSymbol={tokenSymbol}
           isUsingPaymaster={isUsingPaymaster}
+          renderSimulationFailureWarning={renderSimulationFailureWarning}
```
Comment (cryptotavares) [inline ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js:564]: This is not needed anymore

File: ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js [hunk 2]
```diff
@@ -586,7 +572,6 @@ export default class ConfirmTransactionBase extends Component {
             renderSimulationFailureWarning ? null : () => this.handleEditGas()
           }
           rows={[
-            renderSimulationFailureWarning && simulationFailureWarning(),
             !renderSimulationFailureWarning && (
```
Comment (cryptotavares) [inline ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js:575]: if `renderSimulationFailureWarning` is truthy we are rendering an empty `TransactionDetail` component. You need to move the `renderSimulationFailureWarning` validation to wrap the entire TransactionDetail component.

File: ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js [hunk 3]
```diff
@@ -586,7 +572,6 @@ export default class ConfirmTransactionBase extends Component {
             renderSimulationFailureWarning ? null : () => this.handleEditGas()
```
Comment (cryptotavares) [inline ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js:572]: If we move the `renderSimulationFailureWarning` to encapsulate the `TransactionDetail` component, then we can also remove this ternary operator.

Comment (blackdevelopa) [reply ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js]: https://github.com/MetaMask/metamask-extension/pull/23886/commits/cb632f17edb983b176767be12805d9137dc7beea

File: ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js (hunk 2 above)
Comment (blackdevelopa) [reply ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js]: https://github.com/MetaMask/metamask-extension/pull/23886/commits/cb632f17edb983b176767be12805d9137dc7beea

**Review by cryptotavares — APPROVED** [review]

File: ui/pages/confirmations/components/transaction-alerts/transaction-alerts.js [hunk 4]
```diff
@@ -80,7 +80,7 @@ const TransactionAlerts = ({
         />
       )}
 
-      {supportsEIP1559 && hasSimulationError && (
+      {hasSimulationError && (
```
[the remark this answers is not in the document]
Comment (blackdevelopa) [reply ui/pages/confirmations/components/transaction-alerts/transaction-alerts.js:83]: Yes, this should show the error on both fee market and legacy transactions

**Review by jpuri — APPROVED** [review]

**Pull request merged**